### DISCUSSION TITLE
Python SDK

### DISCUSSION THREAD

**Discussion by bobby-didcoding** [hiero-ledger/hiero-improvement-proposals#180] [Ideas]
Python has grown exponentially over the past 10 years and shows no sign of slowing down. It is loved by many developers (including myself) for its simple syntax and fantastic community. This is exactly why I am certain a Python SDK would be welcomed with open arms by us Python developers and therefore encourage more of us to code on Hedera. 

It is worth noting that a community managed SDK has been built and works very well indeed. However, it is basically a Python wrapper for the Java SDK. Therefore we do not have Hedera managed documentation.

Do you think a Python SDK would be as popular as the JAVA, JavaScript and GO SDK’s that we already have?

**Comment by gamilu** [comment]
I second this. I am very much looking forward to a Python SDK. I'm willing to help any way I can to make this happen.

**Reply by bobby-didcoding** [reply]
Glad to hear it! With any luck, this discussion will lead to HIP sign off.

**Comment by wensheng** [comment]
I actually was writing the pure Python SDK before I realized the wrapper approach is quick and easy.  For my current intents and purposes, the Python wrapper SDK works great.  I work mostly in vim in Linux terminal so I don't feel any inconvenience.  But I do realized a lot of, maybe a majority of developers, use an IDE.  In IDE's like PyCharm, Visual studio Code, lack of syntax highlighting, autocompletion, and type hinting really hinder the productivity.  In the wrapper approach, there's no easy way to add those functionalities.  In other words, IDE can not infer Java data types in python code, thus can not give suggestions for syntax highlighting, autocompletion, etc.   This is a serious problem without easy solution.

So yes, I upvote for a pure Python SDK.  

I can even restart working on it in my free time if Hedera doesn't have manpower to spare now.   It's not difficult, just will take a lot of time.

**Reply by bobby-didcoding** [reply]
The wrapper is outstanding!! As you already know, It inspired me to create my Hedera Python tutorials. 
I'm sure Hedera will liaise with you directly if the HIP gets approved.
Keep up the good work

**Reply by mgarbs** [reply]
[Here is the wrapper](https://github.com/wensheng/hedera-sdk-py) for anyone who wants to check it out.

**Comment by 0xJepsen** [comment]
Since python is one of ore popular scripting languages and commonly the first language a lot of new developers learn, I think this would be a great idea!

**Reply by bobby-didcoding** [reply]
You would be hard pushed to find a developer that hasn't at lest dabbled in Python :)
It was the first language that I learnt.

**Comment by a8v0r6** [comment]
Python SDK would be definitely easier to implement. I think it would popularize the Hedera hashgraph in the python dev community as well. 
I cant think of a reason not having a python SDK.

**Reply by bobby-didcoding** [reply]
I agree. Python has one of the best communities out there. They would welcome Hedera with open arms.

**Comment by gorikfr** [comment]
Python SDK would certainly be useful. A lot of machine learning is done in Python and would be nice to have an Hedera SDK in python.

**Reply by bobby-didcoding** [reply]
Yes, I can image there will be some interesting app build using Python.

**Comment by bugbytesinc** [comment]
Perhaps someone from the Python community should submit a proposal to the foundation?

**Reply by bobby-didcoding** [reply]
There is a HIP being put together as we speak 👍

**Comment by 0xJepsen** [comment]
Note a first draft has just been published as a good starting point
https://github.com/hashgraph/hedera-improvement-proposal/pull/257

**Reply by walter-hernandez** [reply]
Thank you for putting it together! It seems what is pending now is to get a reviewer to approve the pull request. If there is anything else needed,  I can give some updates to the HIP if the reviewer rejects the draft

**Reply by 0xJepsen** [reply]
That's good! I think commenting on the hip positively could show community interest and support.

**Reply by walter-hernandez** [reply]
You mean commenting here: https://github.com/hashgraph/hedera-improvement-proposal/pull/257 to show community interest?
